lib/core/test/unit/Cardano/Wallet/Primitive/CollateralSpec.hs

```diff
@@ -0,0 +1,774 @@
+            it "classifies shelley key hash type correctly" $
+                property prop_addressType_shelleyKeyHash
+            it "classifies shelley script hash type correctly" $
+                property prop_addressType_shelleyScriptHash
+            it "golden" $ do
+                unit_addressType_byronGolden
+                unit_addressType_shelleyEnterprisePaymentGolden
+                unit_addressType_stakeAddrGolden
+                unit_addressType_pointerAddrGolden
+                unit_addressType_delegationAddrGolden
+            describe "addressType" $ do
+                it "satisfies same properties as getAddressType" $
+                    property prop_addressType_equivalance
+        describe "collateral suitability" $ do
+            describe "generators and shrinkers" $ do
+                it "generates values with sufficient coverage" $
+                    property prop_genAddress_coverage
+                it "shrink maintains validity" $
+                    property prop_simplifyAddress_validAddress
+                it "shrink maintains type" $
+                    property prop_simplifyAddress_typeMaintained
+            describe "addressSuitableForCollateral" $ do
+                it "assesses all addresses correctly" $
+                    property prop_addressSuitableForCollateral
+                it "golden" $ do
+                    unit_addressSuitableForCollateral_byronGolden
+                    unit_addressSuitableForCollateral_shelleyEnterprisePaymentGolden
+                    unit_addressSuitableForCollateral_stakeAddrGolden
+                    unit_addressSuitableForCollateral_pointerAddrGolden
+                    unit_addressSuitableForCollateral_delegationAddrGolden
+            describe "addressTypeSuitableForCollateral" $ do
+                it "satisfies same properties as addressSuitableForCollateral" $
+                    property prop_addressSuitableForCollateral_equivalence
+            describe "asCollateral" $ do
+                it "satisfies same boolean properties as addressSuitableForCollateral" $
+                    property prop_equivalence_bool
+                it "satisfies same properties as addressSuitableForCollateral" $
+                    property prop_equivalence
+
+-- The following golden keys were generated from the recovery phrase:
+-- [change twin tired knee syrup cover dog glare canvas canvas jump egg]
+
+-- cat recovery-phrase.txt | cardano-address key from-recovery-phrase Byron > root.prv
+-- cat root.prv | cardano-address key child 14H/42H | tee addr.prv | cardano-address key public --with-chain-code | cardano-address address bootstrap --root $(cat root.prv | cardano-address key public --with-chain-code) --network-tag testnet 14H/42H
+byronAddrGolden :: BL.ByteString
+byronAddrGolden = BL.fromStrict . fromJust . decodeBase58 bitcoinAlphabet $ "37btjrVyb4KFsMoVwPRZ5aJko48uBFFUnJ46eV3vC3uBCC65mj5BfbGP6jYDfhojm8MAayHo4RPvWH4x852FcJq8SHazCx31FJM2TfDpV9Azrc8UKD"
```

Comment:
This might make it easier to read in an editor:

```suggestion
byronAddrGolden = BL.fromStrict . fromJust . decodeBase58 bitcoinAlphabet $ 
    "37btjrVyb4KFsMoVwPRZ5aJko48uBFFUnJ46eV3vC3uBCC65mj5BfbGP6jYDfhojm8MAayHo4RPvWH4x852FcJq8SHazCx31FJM2TfDpV9Azrc8UKD"
```